multiline_input.go

```diff
@@ -0,0 +1,101 @@
+package survey
+
+import (
+	"fmt"
+	"strings"
+
+	"gopkg.in/AlecAivazis/survey.v1/core"
+)
+
+type MultilineInput struct {
+	core.Renderer
+	Message string
+	Default string
+	Help    string
+}
+
+// data available to the templates when processing
+type MultilineInputTemplateData struct {
+	MultilineInput
+	Answer     string
+	ShowAnswer bool
+	ShowHelp   bool
+}
+
+// Templates with Color formatting. See Documentation: https://github.com/mgutz/ansi#style-format
+var MultilineInputQuestionTemplate = `
+{{- if .ShowHelp }}{{- color "cyan"}}{{ HelpIcon }} {{ .Help }}{{color "reset"}}{{"\n"}}{{end}}
+{{- color "green+hb"}}{{ QuestionIcon }} {{color "reset"}}
+{{- color "default+hb"}}{{ .Message }} {{color "reset"}}
+{{- if .ShowAnswer}}
+  {{- color "cyan"}}{{.Answer}}{{color "reset"}}{{"\n"}}
+{{- else }}
+  {{- if .Default}}{{color "white"}}({{.Default}}) {{color "reset"}}{{end}}
+  {{- color "cyan"}}[Empty line to finish]{{color "reset"}}
+{{- end}}`
+
+func (i *MultilineInput) Prompt() (interface{}, error) {
+	// render the template
+	err := i.Render(
+		MultilineInputQuestionTemplate,
+		MultilineInputTemplateData{MultilineInput: *i},
+	)
+	if err != nil {
+		return "", err
+	}
+	fmt.Println()
+
+	// start reading runes from the standard in
+	rr := i.NewRuneReader()
+	rr.SetTermMode()
+	defer rr.RestoreTermMode()
+
+	cursor := i.NewCursor()
+
+	multiline := make([]string, 0)
+
+	emptyOnce := false
+	// get the next line
+	for {
+		line := []rune{}
+		line, err = rr.ReadLine(0)
+		if err != nil {
+			return string(line), err
+		}
+
+		if string(line) == "" {
+			if emptyOnce {
+				numLines := len(multiline) + 2
+				cursor.PreviousLine(numLines)
+				for i := 0; i < numLines; i++ {
+					fmt.Println("\033[K")
```

Comment:
I dont think this will work in a windows environment. This is the ansi escape escape sequence for clearing the rest of the line right? I think you need to update this to use `terminal.EraseLine(i.stdio.Out, terminal.ERASE_LINE_END)`

Reply:
\033 == \x1b (which is what `terminal.EraseLine` is doing) but I agree that using your API is clearer.

I've changed it and adjusted to get the same output. I needed to add an extra \n to make it work the same way.

Reply:
I've added also an example for multiline-input in the README, the only thing missing is the gif (I've used the same one as for input).

Don't forget to create a new one @AlecAivazis !